src/api/AppActivity.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * @flow
+ * @file Helper for the box App Activity API
+ * @author Box
+ */
+import MarkerBasedAPI from './MarkerBasedAPI';
+import { ERROR_CODE_DELETE_APP_ACTIVITY } from '../constants';
+import { APP_ACTIVITY_FIELDS_TO_FETCH } from '../utils/fields';
+
+class AppActivity extends MarkerBasedAPI {
+    /**
+     * Map an entry from the AppActivity API to an AppActivityItem.
+     * occurred_at -> created_at
+     *
+     * @param {Object} item - A single entry in the AppActivity API entiries list
+     */
+    mapAppActivityItem(item: Object): AppActivityItem {
+        const { occurred_at, ...rest } = item;
+        return {
+            created_at: occurred_at,
+            ...rest,
+        };
+    }
+
+    /**
+     * Base API URL for App Activity on a file
+     *
+     * @return {string} Base API url for app activities
+     */
+    getAppActivityUrl(): string {
+        return `${this.getBaseApiUrl()}/app_activities`;
+    }
+
+    /**
+     * API URL for getting App Activity on a file
+     *
+     * @param {string} [id] - a box file id
+     * @return {string} Url for all app activity on a file
+     */
+    getUrl(id: string): string {
+        if (!id) {
+            throw new Error('Missing file id!');
+        }
+
+        return `${this.getAppActivityUrl(
+            id,
+        )}?item_id=${id}&item_type=file&fields=${APP_ACTIVITY_FIELDS_TO_FETCH.toString()}`;
```

Comment:
It seems like we should be passing these values an object to `this.markerGet` as part of `requestData` rather than doing string concatenation.

Reply:
That would be a heck of a lot cleaner. Can do!